### ISSUE TITLE
Packagemanager does not show installed libraries by default

### ISSUE THREAD

**Issue by cwittern** [eXist-db/dashboard#130]
> To be able to better understand you problem, please add as much information as possible to this ticket. Always test your bugs against the latest stable release of exist. We cannot provide support for older versions here on GitHub. If the version of eXist that is experiencing the issue is more than 1 major version behind the most recent release, please consider posting a question on our mailing list. 

eXist 5.2.0 on Ubuntu 10.19 in most recent Firefox

**Describe the bug**

The package manager has a panel that displays installed packages.  Only Applications are shown by default.  For the libraries, they have to be made visible by the user through a rather well hidden button in the upper right corner

**Expected behavior**

I would expect all packages to be visible by default as with the old package manager.  If that is not possible *clearly label* the button that makes the libraries visible!

**To Reproduce**

Open the package manager and look for installed libraries.

**Comment by line-o** [comment]
Yes, I agree that the button should be much more prominent (as it was in older versions).
I do want to point out that even those older versions did not show libraries by default. The behaviour has not changed.

**Comment by line-o** [comment]
Would you agree that the issue should be retitled to: "Make it easier to display installed libraries in Packagemanager"?

**Comment by joewiz** [comment]
@line-o Dashboard 1.0 did show library packages by default (see functx in the screenshot below), and it clearly labeled which packages were applications vs. library (see the blue vs. green ribbons displayed prominently):

> ![Dashboard 1.0 showing functx library package](https://gawati.readthedocs.io/en/latest/_images/dashboard.jpg)

The issue is a good articulation of a common question on the forums, so the title and proposal is valid as stated. Certainly it doesn't rule out some creative new solution that isn't an exact replica of the 1.0 functionality, but the current implementation leaves many users scratching their heads.

**Comment by line-o** [comment]
Well, then I am sorry. Must have been using later versions for too long. My mind is still sure that it had to enable libraries even within this layout.
I am not against seeing all that is installed by default. Especially since the launcher does not list libraries.

**Comment by JoernT** [comment]
just my 2 cents:

i decided for hiding libraries back then to avoid scrolling long lists when
looking for an app. I still stick to that argument.

However the option to hide/show libraries can be more prominent AND should
be stored as an user-option (local storage) so people can choose what they
prefer. (that's btw the way i did it in the newer dashboard that
unfortunately waits for an timeslot to be continued)

**Comment by adamretter** [comment]
@JoernT I think it was a nice idea but it hasn't worked in practice. I myself have received at least 5 or 6 enquiries about why libraries are missing or not installed. I suspect I am not alone. This really has to be changed.

**Comment by line-o** [comment]
There is no need to scroll a long list when there is the option to filter the list. This works fine and is way more powerful. The filter input could be placed more prominently, e.g. right on top of the list to make this option pop out.

**Comment by line-o** [comment]
Should we move this issue to https://github.com/eXist-db/packagemanager ?

**Comment by adamretter** [comment]
> This works fine and is way more powerful.

@line-o If we want to provide a good user experience - I think we should listen to our users, rather than assuming that not scrolling makes hiding things a better experience.

We have clearly had several users complain that they can't find libraries - that is undoubtedly a bad user experience. If there is some reason why scrolling is now considered a bad user experience - then we could add paging to fix that. Hiding things is clearly not working!

The most important thing is the user-experience, which should be intuitive and simple. At the moment we don't achieve that.

> The filter input could be placed more prominently, e.g. right on top of the list to make this option pop out.

So are you suggesting a tick-box or something which is always visible? I guess that would be unmissable, which is a good thing.

I would still argue that everything should be shown by default. Otherwise, to assume that apps are more important than libraries is very much assuming that most users install apps over libraries. Most users that I know that use EXPath packaging install many more libraries than apps, they usually have just 1 app (their app) and then rely on lots of libraries.

**Comment by line-o** [comment]
@adamretter yes, exactly my points 
> There is no need to scroll a long list when there is the option to filter the list.
